src/main/java/org/prebid/server/bidder/applogy/ApplogyBidder.java

```diff
@@ -0,0 +1,171 @@
+
+public class ApplogyBidder implements Bidder<BidRequest> {
+
+    private final String endpointUrl;
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    public ApplogyBidder(String endpointUrl) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<Imp> validImps = new ArrayList<>();
+        for (Imp imp : request.getImp()) {
+            if (imp.getBanner() == null && imp.getVideo() == null && imp.getXNative() == null) {
+                errors.add(BidderError.badInput(
+                        String.format("Applogy only supports banner, video and native media types. Ignoring imp id=%s",
+                                imp.getId())));
+                continue;
+            }
+            validImps.add(processImp(imp));
+        }
+
+        if (validImps.isEmpty()) {
+            errors.add(BidderError.badInput("No valid impressions in the bid request"));
+            return Result.of(Collections.emptyList(), errors);
+        }
+
+        final ExtImpApplogy firstImpExt;
+        try {
+            firstImpExt = parseAndValidateImpExt(validImps.get(0));
+        } catch (PreBidException e) {
+            return Result.emptyWithError(BidderError.badInput(e.getMessage()));
+        }
+
+        final BidRequest outgoingRequest = request.toBuilder().imp(validImps).build();
+        final String body = Json.encode(outgoingRequest);
+        final String requestUrl = endpointUrl + "/" + firstImpExt.getToken();
+        final MultiMap headers = resolveHeaders();
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<BidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(requestUrl)
+                        .headers(headers)
+                        .payload(outgoingRequest)
+                        .body(body)
+                        .build()),
+                errors);
+    }
+
+    private MultiMap resolveHeaders() {
+        return HttpUtil.headers();
+    }
+
+    private ExtImpApplogy parseAndValidateImpExt(Imp imp) {
+        final ExtImpApplogy extImpApplogy;
+        try {
+            extImpApplogy = Json.mapper.convertValue(imp.getExt().get("bidder"), ExtImpApplogy.class);
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+
+        if (StringUtils.isBlank(extImpApplogy.getToken())) {
+            throw new PreBidException("token is empty");
+        }
+        return extImpApplogy;
+    }
+
+    private Imp processImp(Imp imp) {
+        Banner banner = imp.getBanner();
+        if (banner != null) {
+            if (banner.getH() == null && banner.getW() == null && CollectionUtils.isNotEmpty(banner.getFormat())) {
```

Comment:
In go they also have a `== 0` check for banner Width and Height, and it should be a `||` (OR) condition, not AND.

Remove `CollectionUtils.isNotEmpty(banner.getFormat())` condition from here. We should check if they ARE empty before getting the first Format, and if they ARE empty - throw the exception - just like they have it in GO.